src/Octoshift/VersionChecker.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace OctoshiftCLI
+{
+    public class VersionChecker
+    {
+        private string _latestVersion;
+
+        public async Task<bool> IsLatest()
+        {
+            var curVersion = GetCurrentVersion();
+            var latestVersion = await GetLatestVersion();
+
+            curVersion = curVersion[..latestVersion.Length];
+
+            return curVersion == latestVersion;
+        }
+
+        public string GetCurrentVersion()
+        {
+            return Assembly.GetExecutingAssembly().GetName().Version.ToString();
+        }
+
+        public async Task<string> GetLatestVersion()
+        {
+            if (string.IsNullOrWhiteSpace(_latestVersion))
```

Comment:
nit: we have an extension method for this now :) --> `_latestVersion.IsNullOrWhiteSpace()`, it reads a bit nicer nothing more :)